qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -227,15 +228,19 @@ def compute_eigenvalues(
         if self.betas is None:
             if isinstance(operator, PauliSumOp):
                 upper_bound = abs(operator.coeff) * sum(
-                    abs(operation.coeff) for operation in operator
+                    abs(operation.primitive.coeffs) for operation in operator
```

Comment:
Could this simplified to the following?
```suggestion
                    np.abs(operator.primitive.coeffs) 
```
Or maybe even just cast to a `SparsePauliOp` for the whole function?
```
if isinstance(operator, PauliSumOp):
    operator = operator.coeff * operator.primitive
```